File: native/android/src/main/java/expo/modules/alarmscheduler/AlarmNotificationCopy.kt
package expo.modules.alarmscheduler

data class AlarmNotificationCopy(
  val title: String,
  val body: String,
  val channelName: String,
)

private const val DEFAULT_ALARM_NOTIFICATION_LANGUAGE = "en"

private val ALARM_NOTIFICATION_COPIES = hashMapOf(
  DEFAULT_ALARM_NOTIFICATION_LANGUAGE to AlarmNotificationCopy(
    title = "Time to wake up!",
    body = "Click to disable the alarm.",
    channelName = "Alarms",
  ),
  "zh-Hant" to AlarmNotificationCopy(
    title = "起床時間到了！",
    body = "輕觸以關閉鬧鐘。",
    channelName = "鬧鐘",
  ),
)

val ENGLISH_ALARM_NOTIFICATION_COPY =
  requireNotNull(ALARM_NOTIFICATION_COPIES[DEFAULT_ALARM_NOTIFICATION_LANGUAGE])

fun alarmNotificationCopy(language: String?): AlarmNotificationCopy =
  language?.let(ALARM_NOTIFICATION_COPIES::get) ?: ENGLISH_ALARM_NOTIFICATION_COPY

File: src/notifications/alarmNotificationCopy.ts
import { i18n } from "@/i18n";

const DEFAULT_ALARM_NOTIFICATION_TITLE = "Time to wake up!";
const DEFAULT_ALARM_NOTIFICATION_BODY = "Click to disable the alarm.";
const DEFAULT_ALARM_NOTIFICATION_CHANNEL_NAME = "Alarms";

export interface AlarmNotificationCopy {
  title: string;
  body: string;
  channelName: string;
}

export function getAlarmNotificationCopy(): AlarmNotificationCopy {
  return {
    title: i18n.t("notifications.title", {
      defaultValue: DEFAULT_ALARM_NOTIFICATION_TITLE,
    }),
    body: i18n.t("notifications.body", {
      defaultValue: DEFAULT_ALARM_NOTIFICATION_BODY,
    }),
    channelName: i18n.t("notifications.channelName", {
      defaultValue: DEFAULT_ALARM_NOTIFICATION_CHANNEL_NAME,
    }),
  };
}
